### DISCUSSION TITLE
What is the exact scope of work for first audit?

### DISCUSSION THREAD

**Discussion by DecentralizedDan** [interlock-network/interlock-smartcontracts#14] [General]
As per our conversation with Solidified / Oak, we need to know exactly what our auditors are going to audit.
* @galactic-beyond @blairmunroakusa please comment below with what work you will deliver for the auditors, whomever they are
* @justerhan please add anything you're going to deliver and let us know if we're leaving anything out!

**Comment by blairmunroakusa** [comment]
What I can speak to, for now:
In terms of our **ERC20 token**, we will deliver an Ethereum/Solidity contract which for now does the following:
- implements ERC20 compliant INTR token
- defines a robust supply mechanism ('mint')
- enforces token emission schedule (via timevault strategy)
- provides cryptographic registration/ claim mechanism for validating approved members/ investors/ whitelisters
( - OR, airdrops to registered members directly, depending on how much $ willing to spend)
- implements standard security guards
- provides slash and burn mechanisms if/when we decide on a concrete spec

~ if we go with the claim mechanism for distribution, we will also provide the signature generation .js script

Wormhole and Solana will become much more apparent in very near future, so I'm personally not going to go there for now.

**Comment by DecentralizedDan** [comment]
See https://github.com/interlock-network/knowledgebase-public/blob/main/what/INTR-smartcontract-stack.md

**Comment by blairmunroakusa** [comment]
UPDATE: THIS DISCUSSION IS OLD AND SOME CONTENTS NO LONGER RELEVANT.

We are building on Aleph Zero. We are no longer building on Ethereum, Wormhole, or Solana.
